### PR TITLE
Add the SquishyTKL and SquishyFRL

### PR DESCRIPTION
**Pull request by ramonimbao**
## Description

Add the SquishyTKL: a gasket o-ring mount TKL keyboard. The SquishyFRL is pretty much the same but without an F-row.

I can't seem to figure out how to use the STM32F103 bootloaders available so I compiled with generic board files instead and flash via ST-Link. If there's a way so I don't have to include those board files with my keyboard files, do let me know. Thank you.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ramonimbao/squishyfrl/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
```
Comment (drashna) [inline keyboards/ramonimbao/squishyfrl/keymaps/via/keymap.c:18]: VIA keymaps require 4 layers.

**Review by drashna — CHANGES_REQUESTED** [review]
Is there a reason for the ChibiOS config files?  There should be defaults in /platform/chibios/,

**Comment by ramonimbao** [comment]
> Is there a reason for the ChibiOS config files? There should be defaults in /platform/chibios/,

I can't get my board to run with the STM32duino board files. I copied my board files from an old project of mine and it worked.

File: keyboards/ramonimbao/squishyfrl/keymaps/via/keymap.c (hunk 1 above)
Comment (ramonimbao) [reply keyboards/ramonimbao/squishyfrl/keymaps/via/keymap.c:18]: Is it okay if I have to adjust the EEPROM memory usage?

Comment (drashna) [reply keyboards/ramonimbao/squishyfrl/keymaps/via/keymap.c:18]: That's fine.  There is a define to lower the number of layers used.  But basically, the number of layers set needs to match the allowed layers, or you get layers filled with junk from the uninitialized EEPROM.  

Specifically, the define that controls this is `DYNAMIC_KEYMAP_LAYER_COUNT`, and defaults to 4. 

Also, changing this means that all users are limited to the number of layers specified.

**Comment by drashna** [comment]
I'm guessing you have different features enabled.  there is a config migration tool built into the `qmk` cli tool that may help convert to the newer stuff. `qmk chibios-confmigrate` specifically.

The main reason I am "making a big deal" about this, is that migrating to a newer version of chibiOS will break your config.  The newer Config that we're using is hardware agnostic, and allows for the main files to be updated, without touching hundreds of keyboards.  Something I'm sure you can understand and appreciate.

**Comment by ramonimbao** [comment]
I managed to get my boards working with the STM32duino bootloader (using `generic_boot20_none.bin`), and flashing with dfu-util. I removed all the generic board files included with my boards now.

Still can't get 4 layers to work. I'll stick with 1 layer for the TKL and 2 layers for the FRL.

File: keyboards/ramonimbao/squishyfrl/keymaps/via/config.h [hunk 2]
```diff
@@ -0,0 +1 @@
+#define DYNAMIC_KEYMAP_LAYER_COUNT 2
```
Comment (drashna) [inline keyboards/ramonimbao/squishyfrl/keymaps/via/config.h:1]: This needs a license head and include guard (`#pragma once`).

File: keyboards/ramonimbao/squishytkl/keymaps/via/config.h [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#define DYNAMIC_KEYMAP_LAYER_COUNT 1
```
Comment (drashna) [inline keyboards/ramonimbao/squishytkl/keymaps/via/config.h:1]: Needs a license header and include guard, too.

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**